erlang: 23.1.4 -> 23.2.6, 22.3 -> 22.3.4.16, 21.3.8.3 -> 21.3.8.21

bobrik opened the pull request:
###### Motivation for this change

Bumping versions to enable `aarch64-darwin` support in #105026.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

bobrik committed:
erlang: add missing AGL framework for aarch64-darwin

Preventing this:

```
 LD     ../priv/aarch64-apple-darwin20.3.0/wxe_driver.so
ld: file not found: /System/Library/Frameworks/AGL.framework/Versions/A/AGL for architecture arm64
clang-11: error: linker command failed with exit code 1 (use -v to see invocation)
make[3]: *** [Makefile:176: ../priv/aarch64-apple-darwin20.3.0/wxe_driver.so] Error 1
```

bobrik committed:
erlangR23: 23.1.4 -> 23.2.6

bobrik committed:
erlangR22: 22.3 -> 22.3.4.16

bobrik committed:
erlangR21: 21.3.8.3 -> 21.3.8.21

happysalada commented:
Result of `nixpkgs-review pr 114589` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages marked as broken and skipped:</summary>
  <ul>
    <li>couchdb</li>
    <li>couchdb2</li>
    <li>couchdb3</li>
    <li>cuter</li>
    <li>erlangR18</li>
    <li>erlangR19</li>
    <li>erlang_basho_R16B02</li>
    <li>lfe_1_2</li>
  </ul>
</details>
<details>
  <summary>21 packages failed to build:</summary>
  <ul>
    <li>asls</li>
    <li>elixir (elixir_1_11)</li>
    <li>elixir_1_10</li>
    <li>elixir_1_7</li>
    <li>elixir_1_8</li>
    <li>elixir_1_9</li>
    <li>erlang (erlangR23)</li>
    <li>erlangR20</li>
    <li>erlangR21</li>
    <li>erlangR22</li>
    <li>erlang_javac</li>
    <li>erlang_nox</li>
    <li>erlang_odbc</li>
    <li>erlang_odbc_javac</li>
    <li>lfe (lfe_1_3)</li>
    <li>mercury</li>
    <li>rabbitmq-server</li>
    <li>rebar</li>
    <li>rebar3</li>
    <li>relxExe</li>
    <li>tsung</li>
  </ul>
</details>

DianaOlympos commented:
Why do we still run erlangR20?

but urgh that seems not normal that they break @happysalada . I do not have a mac, so you will have to have a look at it

SuperSandro2000 commented:
/rebase master

bobrik commented:
The error I'm getting before anything starts to build with `nixpkgs-review`:

```
$ nix --experimental-features nix-command build --no-link --keep-going --option build-use-sandbox relaxed -f /Users/bobrik/.cache/nixpkgs-review/pr-114589/build.nix
error: --- Error --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- nix
getting attributes of path '/usr/lib/libSystem.B.dylib: No such file or directory
(use '--show-trace' to show detailed location information)

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/114589

21 packages failed to build:
asls elixir elixir_1_10 elixir_1_7 elixir_1_8 elixir_1_9 erlang erlangR20 erlangR21 erlangR22 erlang_javac erlang_nox erlang_odbc erlang_odbc_javac lfe mercury rabbitmq-server rebar rebar3 relxExe tsung
```

This doesn't seem like something I broke.

I can build `erlangR22` with `nix-env`:

```
$ nix-env --file . --install --attr erlangR22
installing 'erlang-22.3.4.16'
...
...
```

bobrik commented:
`erlangR22` managed to fail down the road due to missing `Webkit.framework`, looking into that.

SuperSandro2000 commented:
> I can build `erlangR22` with `nix-env`:

Probably sandbox breaks it.

bobrik committed:
erlang: add missing WebKit framework on darwin

Fixing the following error:

```
 LD     ../priv/x86_64-apple-darwin20.3.0/wxe_driver.so
ld: file not found: /System/Library/Frameworks/WebKit.framework/Versions/A/WebKit for architecture x86_64
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
```

bobrik commented:
Same error happens on `master`, huh. I added a commit to add `WebKit`, which fixed `erlangR22` for me.

bobrik commented:
@SuperSandro2000 indeed: https://github.com/Mic92/nixpkgs-review/commit/aa9cc2f5e9199736af5167f8c3f406018b6a14e7.

bobrik commented:
The only broken thing at the latest commit is `tsung`, but it's also broken in `master` due to Python 2.7:

* https://hydra.nixos.org/job/nixpkgs/trunk/tsung.x86_64-darwin

happysalada commented:
Result of `nixpkgs-review pr 114589` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages marked as broken and skipped:</summary>
  <ul>
    <li>couchdb</li>
    <li>couchdb2</li>
    <li>couchdb3</li>
    <li>cuter</li>
    <li>erlangR18</li>
    <li>erlangR19</li>
    <li>erlang_basho_R16B02</li>
    <li>lfe_1_2</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>tsung</li>
  </ul>
</details>
<details>
  <summary>20 packages built:</summary>
  <ul>
    <li>asls</li>
    <li>elixir (elixir_1_11)</li>
    <li>elixir_1_10</li>
    <li>elixir_1_7</li>
    <li>elixir_1_8</li>
    <li>elixir_1_9</li>
    <li>erlang (erlangR23)</li>
    <li>erlangR20</li>
    <li>erlangR21</li>
    <li>erlangR22</li>
    <li>erlang_javac</li>
    <li>erlang_nox</li>
    <li>erlang_odbc</li>
    <li>erlang_odbc_javac</li>
    <li>lfe (lfe_1_3)</li>
    <li>mercury</li>
    <li>rabbitmq-server</li>
    <li>rebar</li>
    <li>rebar3</li>
    <li>relxExe</li>
  </ul>
</details>

SuperSandro2000 commented:
> @SuperSandro2000 indeed: [Mic92/nixpkgs-review@aa9cc2f](https://github.com/Mic92/nixpkgs-review/commit/aa9cc2f5e9199736af5167f8c3f406018b6a14e7).

I am using nixpkgs-reviews master so this is not happening for me.

bobrik commented:
Any blockers for merging this?

happysalada commented:
I'm for merging. @yurrriq when you have a moment, perhaps you can take a look at this?

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 114589` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package blacklisted:
</summary>
<ul>
<li>tests.nixos-functions.nixos-test</li>
</ul>
</details>
<details>

  <summary>1 package failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>tsung: log was empty</li>
  </ul>
</details>
<details>
<summary>
25 packages built:
</summary>
<ul>
<li>asls</li>
<li>cl</li>
<li>couchdb2</li>
<li>couchdb3</li>
<li>ejabberd</li>
<li>elixir (elixir_1_11)</li>
<li>elixir_1_10</li>
<li>elixir_1_7</li>
<li>elixir_1_8</li>
<li>elixir_1_9</li>
<li>erlang (erlangR23)</li>
<li>erlangR21</li>
<li>erlangR22</li>
<li>erlang_javac</li>
<li>erlang_nox</li>
<li>erlang_odbc</li>
<li>erlang_odbc_javac</li>
<li>lfe (lfe_1_3)</li>
<li>mercury</li>
<li>notmuch-bower</li>
<li>rabbitmq-server</li>
<li>rebar</li>
<li>rebar3</li>
<li>relxExe</li>
<li>wings</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
cl:
<details>


warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/libraries/cl/default.nix:28:3:

```
   |
28 |   meta = with lib; {
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
couchdb2:
<details>


warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/servers/http/couchdb/2.0.0.nix:55:5:

```
   |
55 |     maintainers = with maintainers; [ ];
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
couchdb3:
<details>


warning: unused-argument
Unused argument: `coreutils`.
Near pkgs/servers/http/couchdb/3.nix:2:3:

```
  |
2 | , coreutils, bash, makeWrapper, python3 }:
  |   ^
```
warning: unused-argument
Unused argument: `bash`.
Near pkgs/servers/http/couchdb/3.nix:2:14:

```
  |
2 | , coreutils, bash, makeWrapper, python3 }:
  |              ^
```
warning: unused-argument
Unused argument: `makeWrapper`.
Near pkgs/servers/http/couchdb/3.nix:2:20:

```
  |
2 | , coreutils, bash, makeWrapper, python3 }:
  |                    ^
```
</details>
ejabberd:
<details>


warning: unclear-gpl
`gpl2` is a deprecated license, check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/servers/xmpp/ejabberd/default.nix:115:5:

```
    |
115 |     license = licenses.gpl2;
    |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
erlang_javac:
<details>


warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern 'temp > NULL' doesn't match anything in file 'lib/wx/c_src/wxe_impl.cpp'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '`sw_vers -productVersion`' doesn't match anything in file 'make/configure.in'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '-Wl,-no_weak_imports' doesn't match anything in file 'erts/configure.in'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
</details>
erlang:
<details>


warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern 'temp > NULL' doesn't match anything in file 'lib/wx/c_src/wxe_impl.cpp'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '`sw_vers -productVersion`' doesn't match anything in file 'make/configure.in'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '-Wl,-no_weak_imports' doesn't match anything in file 'erts/configure.in'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
</details>
erlang_nox:
<details>


warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern 'temp > NULL' doesn't match anything in file 'lib/wx/c_src/wxe_impl.cpp'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '`sw_vers -productVersion`' doesn't match anything in file 'make/configure.in'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '-Wl,-no_weak_imports' doesn't match anything in file 'erts/configure.in'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
</details>
erlang_odbc_javac:
<details>


warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern 'temp > NULL' doesn't match anything in file 'lib/wx/c_src/wxe_impl.cpp'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '`sw_vers -productVersion`' doesn't match anything in file 'make/configure.in'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '-Wl,-no_weak_imports' doesn't match anything in file 'erts/configure.in'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
</details>
erlang_odbc:
<details>


warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern 'temp > NULL' doesn't match anything in file 'lib/wx/c_src/wxe_impl.cpp'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '`sw_vers -productVersion`' doesn't match anything in file 'make/configure.in'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '-Wl,-no_weak_imports' doesn't match anything in file 'erts/configure.in'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
</details>
erlangR21:
<details>


warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern 'temp > NULL' doesn't match anything in file 'lib/wx/c_src/wxe_impl.cpp'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '`sw_vers -productVersion`' doesn't match anything in file 'configure.in'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
</details>
erlangR22:
<details>


warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern 'temp > NULL' doesn't match anything in file 'lib/wx/c_src/wxe_impl.cpp'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '`sw_vers -productVersion`' doesn't match anything in file 'make/configure.in'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '-Wl,-no_weak_imports' doesn't match anything in file 'erts/configure.in'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
</details>
lfe:
<details>


warning: attribute-typo
A likely typo in `buildPlugins` argument was found, did you mean `buildFlags`?

Near pkgs/development/beam-modules/build-rebar3.nix:44:5:

```
   |
44 |     buildPlugins = ownPlugins;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/attribute-typo.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '/usr/bin/env' doesn't match anything in file '/nix/store/f3m3zr1v23lyxd3ma2s3vl6jhzq4z7bc-lfe-1.3/bin/lfe'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '/usr/bin/env' doesn't match anything in file '/nix/store/f3m3zr1v23lyxd3ma2s3vl6jhzq4z7bc-lfe-1.3/bin/lfec'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '/usr/bin/env' doesn't match anything in file '/nix/store/f3m3zr1v23lyxd3ma2s3vl6jhzq4z7bc-lfe-1.3/bin/lfedoc'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '/usr/bin/env' doesn't match anything in file '/nix/store/f3m3zr1v23lyxd3ma2s3vl6jhzq4z7bc-lfe-1.3/bin/lfescript'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
</details>
mercury:
<details>

build log of '/nix/store/4yva8k2hv140457xdvyiymaskgvwpb7p-mercury-20.06.1' is not available
error: --- Error --- nix
build log of '/nix/store/4yva8k2hv140457xdvyiymaskgvwpb7p-mercury-20.06.1' is not available
When evaluating attribute ‘mercury’:
warning: unclear-gpl
`gpl2` is a deprecated license, check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/development/compilers/mercury/default.nix:58:5:

```
   |
58 |     license     = lib.licenses.gpl2;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
warning: patch-phase
`patchPhase` should not be overridden, use `postPatch` instead.

Near pkgs/development/compilers/mercury/default.nix:16:3:

```
   |
16 |   patchPhase = ''
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/patch-phase.md
warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/compilers/mercury/default.nix:60:5:

```
   |
60 |     maintainers = [ ];
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
notmuch-bower:
<details>


warning: unclear-gpl
`gpl3` is a deprecated license, check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/applications/networking/mailreaders/notmuch-bower/default.nix:33:5:

```
   |
33 |     license = licenses.gpl3;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
rabbitmq-server:
<details>


warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '`dirname $0`/rabbitmq-env' doesn't match anything in file '/nix/store/k094jzvlvdlknkkadzlvrr6ndx6m0dn0-rabbitmq-server-3.8.9/sbin/rabbitmq-server'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
</details>
relxExe:
<details>


warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/beam-modules/rebar3-release.nix:73:5:

```
   |
73 |     meta = {
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
warning: license-missing
Package is missing a license

Near pkgs/development/beam-modules/rebar3-release.nix:73:5:

```
   |
73 |     meta = {
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/license-missing.md
warning: unused-argument
Unused argument: `enableDebugInfo`.
Near pkgs/development/beam-modules/rebar3-release.nix:15:3:

```
   |
15 | , enableDebugInfo ? false
   |   ^
```
</details>
tsung:
<details>


warning: unclear-gpl
`gpl2` is a deprecated license, check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/applications/networking/tsung/default.nix:46:5:

```
   |
46 |     license = licenses.gpl2;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
wings:
<details>


warning: patch-phase
`patchPhase` should not be overridden, use `postPatch` instead.

Near pkgs/applications/graphics/wings/default.nix:12:3:

```
   |
12 |   patchPhase = ''
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/patch-phase.md
</details>
</details>

SuperSandro2000 commented:
> I'm for merging. @yurrriq when you have a moment, perhaps you can take a look at this?

We should address the stale substituteInPlace.

bobrik committed:
erlangR23, erlangR22, erlangR21: remove obsolete substituteInPlace

bobrik commented:
> We should address the stale substituteInPlace.

Done.

happysalada commented:
@SuperSandro2000 nice improvement over nixpkgs-review!
It looks good now, what do you think?

yurrriq commented:
Smoke testing on NixOS locally:
- [ ] R21
- [x] R22
- [ ] R23

yurrriq commented:
lgtm on NixOS. Any outstanding issues, @SuperSandro2000?

SuperSandro2000 commented:
build failure on darwin:
```
builder for '/nix/store/yri62xycqh3xi1f1kjkh9mlv6d40fgs6-erlang-20.3.8.9.drv' failed with exit code 1; last 10 log lines:
  checking for env... /usr/bin/env
  checking for GNU make... yes (make)
  checking for a BSD-compatible install... /nix/store/cpvjym13fdglv6zmdr5xav20g5rbafbx-coreutils-8.32/bin/install -c
  checking whether ln -s works... yes
  checking for ranlib... ranlib
  checking ERTS version... 9.3.3.3
  checking OTP release... 20
  checking OTP version... 20.3.8.9
  ./configure: line 3902: sw_vers: command not found
  configure: error: Failed to execute 'sw_vers'; please provide it in PATH
```

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 114589` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
9 packages marked as broken and skipped:
</summary>
<ul>
<li>couchdb</li>
<li>cuter</li>
<li>erlangR18</li>
<li>erlangR19</li>
<li>erlang_basho_R16B02</li>
<li>lfe_1_2</li>
<li>riak</li>
<li>tsung</li>
<li>yaws</li>
</ul>
</details>
<details>
<summary>
1 package blacklisted:
</summary>
<ul>
<li>tests.nixos-functions.nixos-test</li>
</ul>
</details>
<details>
<summary>
26 packages built:
</summary>
<ul>
<li>asls</li>
<li>cl</li>
<li>couchdb2</li>
<li>couchdb3</li>
<li>ejabberd</li>
<li>elixir (elixir_1_11)</li>
<li>elixir_1_10</li>
<li>elixir_1_7</li>
<li>elixir_1_8</li>
<li>elixir_1_9</li>
<li>erlang (erlangR23)</li>
<li>erlangR20</li>
<li>erlangR21</li>
<li>erlangR22</li>
<li>erlang_javac</li>
<li>erlang_nox</li>
<li>erlang_odbc</li>
<li>erlang_odbc_javac</li>
<li>lfe (lfe_1_3)</li>
<li>mercury</li>
<li>notmuch-bower</li>
<li>rabbitmq-server</li>
<li>rebar</li>
<li>rebar3</li>
<li>relxExe</li>
<li>wings</li>
</ul>
</details>

bobrik commented:
Fixed `erlangR20` on `x86_64-darwin`.

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 114589` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
9 packages marked as broken and skipped:
</summary>
<ul>
<li>couchdb</li>
<li>couchdb2</li>
<li>couchdb3</li>
<li>cuter</li>
<li>erlangR18</li>
<li>erlangR19</li>
<li>erlang_basho_R16B02</li>
<li>lfe_1_2</li>
<li>tsung</li>
</ul>
</details>
<details>

  <summary>1 package failed to build and are new build failure:</summary>
  <ul>
<li>erlangR20: <a href="https://termbin.com/41mk">plain log</a>  |  <a href="https://htmlpreview.github.io/?https://gist.githubusercontent.com/SuperSandroBot/27d898446e67c2de8dd11e72139c1fec/raw/index.html">pretty log </a></li>
  </ul>
</details>
<details>
<summary>
19 packages built:
</summary>
<ul>
<li>asls</li>
<li>elixir (elixir_1_11)</li>
<li>elixir_1_10</li>
<li>elixir_1_7</li>
<li>elixir_1_8</li>
<li>elixir_1_9</li>
<li>erlang (erlangR23)</li>
<li>erlangR21</li>
<li>erlangR22</li>
<li>erlang_javac</li>
<li>erlang_nox</li>
<li>erlang_odbc</li>
<li>erlang_odbc_javac</li>
<li>lfe (lfe_1_3)</li>
<li>mercury</li>
<li>rabbitmq-server</li>
<li>rebar</li>
<li>rebar3</li>
<li>relxExe</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
asls:
<details>

warning: EvalError
Cannot evaluate attribute ‘asls’ in ‘/nix/store/ll4jzpnp99mxqiqmbb1yys6lz18h046x-nixpkgs’.
</details>
elixir:
<details>

warning: EvalError
Cannot evaluate attribute ‘elixir’ in ‘/nix/store/ll4jzpnp99mxqiqmbb1yys6lz18h046x-nixpkgs’.
</details>
elixir_1_10:
<details>

warning: EvalError
Cannot evaluate attribute ‘elixir_1_10’ in ‘/nix/store/ll4jzpnp99mxqiqmbb1yys6lz18h046x-nixpkgs’.
</details>
elixir_1_7:
<details>

warning: EvalError
Cannot evaluate attribute ‘elixir_1_7’ in ‘/nix/store/ll4jzpnp99mxqiqmbb1yys6lz18h046x-nixpkgs’.
</details>
elixir_1_8:
<details>

warning: EvalError
Cannot evaluate attribute ‘elixir_1_8’ in ‘/nix/store/ll4jzpnp99mxqiqmbb1yys6lz18h046x-nixpkgs’.
</details>
elixir_1_9:
<details>

warning: EvalError
Cannot evaluate attribute ‘elixir_1_9’ in ‘/nix/store/ll4jzpnp99mxqiqmbb1yys6lz18h046x-nixpkgs’.
</details>
erlang:
<details>

warning: EvalError
Cannot evaluate attribute ‘erlang’ in ‘/nix/store/ll4jzpnp99mxqiqmbb1yys6lz18h046x-nixpkgs’.
</details>
erlangR20:
<details>

warning: EvalError
Cannot evaluate attribute ‘erlangR20’ in ‘/nix/store/ll4jzpnp99mxqiqmbb1yys6lz18h046x-nixpkgs’.
</details>
erlangR21:
<details>

warning: EvalError
Cannot evaluate attribute ‘erlangR21’ in ‘/nix/store/ll4jzpnp99mxqiqmbb1yys6lz18h046x-nixpkgs’.
</details>
erlangR22:
<details>

warning: EvalError
Cannot evaluate attribute ‘erlangR22’ in ‘/nix/store/ll4jzpnp99mxqiqmbb1yys6lz18h046x-nixpkgs’.
</details>
erlang_javac:
<details>

warning: EvalError
Cannot evaluate attribute ‘erlang_javac’ in ‘/nix/store/ll4jzpnp99mxqiqmbb1yys6lz18h046x-nixpkgs’.
</details>
erlang_nox:
<details>

warning: EvalError
Cannot evaluate attribute ‘erlang_nox’ in ‘/nix/store/ll4jzpnp99mxqiqmbb1yys6lz18h046x-nixpkgs’.
</details>
erlang_odbc:
<details>

warning: EvalError
Cannot evaluate attribute ‘erlang_odbc’ in ‘/nix/store/ll4jzpnp99mxqiqmbb1yys6lz18h046x-nixpkgs’.
</details>
erlang_odbc_javac:
<details>

warning: EvalError
Cannot evaluate attribute ‘erlang_odbc_javac’ in ‘/nix/store/ll4jzpnp99mxqiqmbb1yys6lz18h046x-nixpkgs’.
</details>
lfe:
<details>

warning: EvalError
Cannot evaluate attribute ‘lfe’ in ‘/nix/store/ll4jzpnp99mxqiqmbb1yys6lz18h046x-nixpkgs’.
</details>
mercury:
<details>

warning: EvalError
Cannot evaluate attribute ‘mercury’ in ‘/nix/store/ll4jzpnp99mxqiqmbb1yys6lz18h046x-nixpkgs’.
</details>
rabbitmq-server:
<details>

warning: EvalError
Cannot evaluate attribute ‘rabbitmq-server’ in ‘/nix/store/ll4jzpnp99mxqiqmbb1yys6lz18h046x-nixpkgs’.
</details>
rebar:
<details>

warning: EvalError
Cannot evaluate attribute ‘rebar’ in ‘/nix/store/ll4jzpnp99mxqiqmbb1yys6lz18h046x-nixpkgs’.
</details>
rebar3:
<details>

warning: EvalError
Cannot evaluate attribute ‘rebar3’ in ‘/nix/store/ll4jzpnp99mxqiqmbb1yys6lz18h046x-nixpkgs’.
</details>
relxExe:
<details>

warning: EvalError
Cannot evaluate attribute ‘relxExe’ in ‘/nix/store/ll4jzpnp99mxqiqmbb1yys6lz18h046x-nixpkgs’.
</details>
</details>

SuperSandro2000 commented on pkgs/development/interpreters/erlang/generic-builder.nix:
```diff
@@ -57,17 +57,14 @@ in stdenv.mkDerivation ({
   prePatch = ''
-    substituteInPlace lib/wx/c_src/wxe_impl.cpp --replace 'temp > NULL' 'temp != NULL'
-
     ${prePatch}
```
Can we replace this with inherit prePatch?

bobrik replied:
Sure, done.

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 114589` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
9 packages marked as broken and skipped:
</summary>
<ul>
<li>couchdb</li>
<li>cuter</li>
<li>erlangR18</li>
<li>erlangR19</li>
<li>erlang_basho_R16B02</li>
<li>lfe_1_2</li>
<li>riak</li>
<li>tsung</li>
<li>yaws</li>
</ul>
</details>
<details>
<summary>
1 package blacklisted:
</summary>
<ul>
<li>tests.nixos-functions.nixos-test</li>
</ul>
</details>
<details>
<summary>
26 packages built:
</summary>
<ul>
<li>asls</li>
<li>cl</li>
<li>couchdb2</li>
<li>couchdb3</li>
<li>ejabberd</li>
<li>elixir (elixir_1_11)</li>
<li>elixir_1_10</li>
<li>elixir_1_7</li>
<li>elixir_1_8</li>
<li>elixir_1_9</li>
<li>erlang (erlangR23)</li>
<li>erlangR20</li>
<li>erlangR21</li>
<li>erlangR22</li>
<li>erlang_javac</li>
<li>erlang_nox</li>
<li>erlang_odbc</li>
<li>erlang_odbc_javac</li>
<li>lfe (lfe_1_3)</li>
<li>mercury</li>
<li>notmuch-bower</li>
<li>rabbitmq-server</li>
<li>rebar</li>
<li>rebar3</li>
<li>relxExe</li>
<li>wings</li>
</ul>
</details>
